pkg/controller/psbackup/controller.go

```diff
@@ -618,10 +619,12 @@ func (r *PerconaServerMySQLBackupReconciler) deleteBackup(ctx context.Context, c
 		}
 		return complete, nil
 	}
-	src, err := r.getBackupSource(ctx, cr, cluster)
+
+	pod, err := ps.GetReadyMySQLPod(ctx, r.Client, cluster)
```

Comment:
not sure if I like controllers depending on each other. I think we should move `GetReadyMySQLPod` into mysql package. like I did in https://github.com/percona/percona-server-mysql-operator/pull/1038/files#diff-3f6060cfa02d4882b8d9603ba089553953c7c32926aa17941bff7f096db7e8bb